lua/apg/modules/tools.lua

```diff
@@ -12,31 +12,29 @@
 local mod = "tools"
 
 function APG.canTool( ply, tool, ent )
-	if not IsValid(ent) then
-		return false
+	if IsValid(ent) then
+		if ent.ToolDisabled == false then
+			return false
+		end
+
+		if ent.CPPICanTool then
+			return ent:CPPICanTool(ply, tool)
+		end -- Let CPPI handle things from here.
+
 	end
 
-	if ent.ToolDisabled == false then
+	if APG.cfg[ "checkCanTool" ].value and ply.APG_CantPickup == true then-- If we can't pickup we can't tool either.
 		return false
 	end
 
-	if ply.APG_CantPickup == true then
-		return false 
-	end -- If we can't pickup we can't tool either.
-
-	if ent.CPPICanTool then 
-		return ent:CPPICanTool(ply, tool)
-	end -- Let CPPI handle things from here.
-
-	return false
+	return 0 -- return 0 so if all of the check's don't return anything then it doesn't default to disabling toolgun.
```

Comment:
If you don't intend to take action, couldn't you just `return nil` or `return`?

Reply:
No, because nil is considered false, and not returning anything does the same, look at APG_ToolMain

Reply:
I thought this was being attached to the hook `CanTool`, which like every other hook won't act if `nil` is passed, which seemed like the desired behavior.

If that's not the case, my bad.